aten/src/THC/THCTensorMathPointwise.cuh

```diff
@@ -1,6 +1,7 @@
 #ifndef THC_TENSORMATH_POINTWISE_CUH
 #define THC_TENSORMATH_POINTWISE_CUH
 
+#include <type_traits>
```

Comment:
Do we add a hard C++ requirement to THC?

Reply:
idk, but we are using static conditional asserts, which needed type_traits